.github/workflows/fork-cherry-pick.yml

```diff
@@ -0,0 +1,46 @@
+on:
+  #Set to trigger on every merge to main, not just a closed PR.
+  workflow_dispatch:
+  pull_request_target:
+    branches:
+      - main
+    types: ["closed"]
+
+jobs:
+  cherry_pick:
+    runs-on: ubuntu-latest
+    name: Cherry pick into main
+    if: github.event.pull_request.merged == true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - name: Create PR Patch Branch
+        shell: bash
+        env: 
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+            git config --global user.name "${GITHUB_ACTOR}"
+            git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+            git remote add upstream https://github.com/cosmos/cosmos-sdk.git
+            git fetch --all                                   # Get the latest code
+            git checkout -b pr-patch upstream/main    # Create new branch based on main branch
+            git cherry-pick ${{github.event.pull_request.head.sha}}  # Cherry pick the latest commit of PR
```

Comment:
@xBalbinus I'm not understanding, but this doesn't seem like the correct approach to me. Shouldn't we be using a single GH API key? Why are we settings configs in CI?

Reply:
So, there are actually two things happening here - The first step is to create a pr-specific branch and cherry pick the head of main (since this is happening after a PR on the fork gets merged into main) into that pr branch, then the GitHub API is used for creating an automatic pull request off of that cherry pick. I hope that explains a bit? Lmk if that doesn't make sense still, happy to revise.

Reply:
Yeah, this doesn't seem right to me. I don't feel like setting configs is the correct approach here. Doesn't the workflow do this for us or something similar?

Reply:
@alexanderbez git config stuff removed, still works as usual.